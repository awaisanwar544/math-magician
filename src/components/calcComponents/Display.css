.display-container {
  display: flex;
  flex-direction: column;
  background-color: #2b2b2b;
  color: #fff;
}

.value {
  line-height: 2em;
  align-self: flex-end;
  padding-right: 1em;
}
