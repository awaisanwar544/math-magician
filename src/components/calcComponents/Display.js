import PropTypes from 'prop-types';
import './Display.css';

const Display = (props) => {
  const { value } = props;
  return (
    <div className="display-container">
      <p className="value">{value}</p>
    </div>
  );
};

Display.defaultProps = { value: 0 };
Display.propTypes = { value: PropTypes.number };

export default Display;
