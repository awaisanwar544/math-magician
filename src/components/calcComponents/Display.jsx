import PropTypes from 'prop-types';
import './Display.css';

const Display = ({ value }) => (
  <div className="display-container">
    <p className="value" id="display-value">{value}</p>
  </div>
);

Display.propTypes = { value: PropTypes.string.isRequired };

export default Display;
