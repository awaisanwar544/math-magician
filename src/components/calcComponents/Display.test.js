import React from 'react';
import renderer from 'react-test-renderer';

import Display from './Display';

describe('Displays Correct Result', () => {
  it('Displays correct answer', () => {
    const tree = renderer
      .create(<Display value="70" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
