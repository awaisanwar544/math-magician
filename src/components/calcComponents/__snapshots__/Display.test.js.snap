// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays Correct Result Displays correct answer 1`] = `
<div
  className="display-container"
>
  <p
    className="value"
    id="display-value"
  >
    70
  </p>
</div>
`;
